modules/core/src/main/java/org/apache/ignite/internal/processors/task/GridTaskProcessor.java

```
@@ -663,8 +664,13 @@ else if (task != null) {
                             // Start task execution in another thread.
                             if (sys)
                                 ctx.getSystemExecutorService().execute(taskWorker);
-                            else
-                                ctx.getExecutorService().execute(taskWorker);
+                            else {
+                                final Object executerName = map.get(EXECUTOR_NAME);
+                                if (executerName != null && executerName instanceof String)
```

do we really need to check for String class belonging here? May be u should cast it above or check in assert with reasonable message.
In case executorName is not null and instance of Foo class, invalid variable type will be ignored(which is undesireable) and normal flow would proceed. So, fail fast rocks !

---

Yep, you are right we don't need class cast here because we always put String in this entry.